### PR TITLE
Fix calendar on blur validation #6415

### LINKED ISSUE #6415: When the form validation is type of blur , then for the <DateInput />  , we can’t select two different dates
When the form validation is type of blur , then for the <DateInput format="mm/dd/yyyy-mm/dd/yyyy"/>  , we can’t select two different dates ! But it works fine for   validation is type of submit . Is this something expected ? 
### Expected Behavior

When the form validation is type of blur , then for the <DateInput format="mm/dd/yyyy-mm/dd/yyyy"/>  , we  should be able to select two different  dates !

### Actual Behavior

When the form validation is type of blur , then for the <DateInput format="mm/dd/yyyy-mm/dd/yyyy"/>  , we can’t select two different dates 

### URL, screen shot, or Codepen exhibiting the issue
https://codesandbox.io/s/dateinput-controlled-forked-t9qf06?file=/index.js

### Steps to Reproduce

1.Create a form with validation type as blur
2. Add a DateInput component with format as  'mm/dd/yyyy-mm/dd/yyyy'
3.  Try selecting two different dates of different months (25th September  2022 and 25th  December 2022)

### Your Environment

- Grommet version:  2.26.0
- Browser Name and version: chrome (106.0.5249.91 (Official Build) (arm64)
- Operating System and version (desktop or mobile): desktop

### PR DESCRIPTION
**Pull request by marcioflaviof**
#### What does this PR do?
Fixes #6415 

#### Where should the reviewer start?
DateInput component

#### What testing has been done on this PR?
 Should be done in E2E tests

#### How should this be manually tested?
Change form to this format 'mm/dd/yyyy-mm/dd/yyyy' and validation to blur

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

#### What are the relevant issues?
#6415 

#### Do the grommet docs need to be updated?
No

#### Should this PR be mentioned in the release notes?
I think no

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible

### CODE REVIEW

File: src/js/components/DateInput/__tests__/DateInput-test.tsx [hunk 1]
```diff
@@ -899,6 +899,51 @@ describe('DateInput', () => {
     expect(dateInputValue).toEqual('07/02/2020-07/07/2020');
   });
 
+  test('should be working with blur validation', async () => {
```
Comment (jcfilben) [inline src/js/components/DateInput/__tests__/DateInput-test.tsx:902]: Just a note, this test is working even without the fix in DateInput. The issue may not be reproducing in the jest testing environment and may be something we want to consider testing in an end to end environment. For now I am okay with leaving this test in and just adding a note in our end to end testing issue that we should replace this with an end to end test.

**Review by jcfilben — APPROVED** [review]

**Comment by jcfilben** [comment]
Actually it looks like the jest test is not passing, I think we can go ahead and just remove the test since it isn't accurately testing the on blur issue anyway

**Comment by marcioflaviof** [comment]
Ok, I'll remove and push

**Review by jcfilben — APPROVED** [review]
Looks good!

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**